docs/evm/cadence/batched-evm-transactions.md

```diff
@@ -0,0 +1,641 @@
+to wrap. The mint amount for the `MaybeMintERC721` contract is 1 whole FLOW which in EVM terms is `1e18 wei` - `wei`
+being the smallest unit of an EVM's native currency (inherited from Ethereum's units - more on Ether units
+[here](https://www.cyfrin.io/glossary/ether-and-wei-code-example)).
+
+Your input field should have `1 000 000 000 000 000 000` in it as shown below.
+
+![Deposit 1 FLOW to WFLOW contract](./wflow-deposit.png)
+
+You can now click the "Write" button to submit the transaction. Once MetaMask prompts you to sign the transaction, click
+"Confirm" and give it a few seconds to process.
+
+![Confirm WFLOW deposit in MetaMask](./wflow-deposit-confirm.png)
+
+Once confirmed, you should be able to see WFLOW balance in your tokens list in MetaMask - if not, you can click on
+"Import Tokens" and paste the WFLOW contract address found on the Flowscan page and refresh your list.
+
+![WFLOW in MetaMask](./wflow-in-metamask-tokens.png)
+
+#### 2. Approve WFLOW Transfer
+
+Now that you have your WFLOW, you'll need to approve the `MaybeMintERC721` contract to transfer your WFLOW. From the
+same WFLOW page in Flowscan, click on the `approve` method. This time, you'll need to input the `MaybeMintERC721`
+contract address - `0x2E2Ed0Cfd3AD2f1d34481277b3204d807Ca2F8c2` - and the amount of WFLOW you want to approve - again `1
+000 000 000 000 000 000` WFLOW.
+
+![Approve MaybeMintERC721 for 1 WFLOW in Flowscan](./wflow-approve.png)
+
+Click "Write" to submit the transaction. To be clear, this does not complete a transfer, but allows the
+`MaybeMintERC721` contract to transfer your WFLOW on your behalf which will execute in the next step.
+
+#### 3. Mint ERC721 Token
+
+Finally, we'll attempt to mint the ERC721 token using the `MaybeMintERC721` contract. Navigate to the `MaybeMintERC721`
+contract on Flowscan:
+[MaybeMintERC721](https://evm-testnet.flowscan.io/address/0x2E2Ed0Cfd3AD2f1d34481277b3204d807Ca2F8c2?tab=write_contract).
+
+Again, you'll be met with the contract functions on the "Write Contract" tab. Click on the `mint` function which takes
+no arguments - just click on "Write" and then "Confirm" in the resulting MetaMask window.
+
+This contract has a 50% chance of failing on mint using onchain randomness. If it fails, simply mint again until it
+succeeds.
+
+On success, you can click on your NFTs in MetaMask to see your newly minted token.
+
+![MaybeMintERC721 in MetaMask NFT list](./maybe-mint-in-metamask.png)
+
+#### Recap
+
+This process is cumbersome and requires multiple transactions, each of which could fail. Given the intent of the
+process - minting an NFT - if this were a case where the NFT was a limited edition or time-sensitive, you'd be left with
+WFLOW wrapped and approved for transfer, but no NFT and would need to manually unwind the process.
+
+Or you could just use Cadence to batch these transactions and revert everything if the mint fails. Let's do that.
+
+### Using Flow Wallet
```

Comment:
H2?

Reply:
As is, the outline works out to be

```markdown
- [Objectives](#objectives)
- [Prerequisites](#prerequisites)
- [Overview](#overview)
- [Components](#components)
  - [Wrap FLOW as WFLOW](#wrap-flow-as-wflow)
  - [Approve ERC721 Transfer](#approve-erc721-transfer)
  - [Mint ERC721 Token](#mint-erc721-token)
- [Interacting with the Contracts](#interacting-with-the-contracts)
  - [Using MetaMask](#using-metamask)
  - [Using Flow Wallet](#using-flow-wallet)
- [Breaking it Down](#breaking-it-down)
  - [COA Configuration](#coa-configuration)
  - [Funding the COA](#funding-the-coa)
  - [Setting our EVM Contract Targets](#setting-our-evm-contract-targets)
  - [Wrapping FLOW as WFLOW](#wrapping-flow-as-wflow)
  - [Approving the ERC721 Contract](#approving-the-erc721-contract)
  - [Minting the ERC721 Token](#minting-the-erc721-token)
- [Conclusion](#conclusion)
- [Further Reading](#further-reading)
```

I think "Using Flow Wallet" makes sense in the same section as "Using MetaMask" unless there's objection.